components/script/dom/htmlbodyelement.rs

```diff
@@ -145,7 +145,9 @@ impl VirtualMethods for HTMLBodyElement {
             atom!("bgcolor") |
             atom!("text") => AttrValue::from_legacy_color(value.into()),
             atom!("background") => {
-                AttrValue::from_url(document_from_node(self).url(), value.into())
+                let doc = document_from_node(self);
+                let val = AttrValue::from_url(&*doc.url(), value.into());
+                val 
```

Comment:
This creates 2 temporary stack variables for the sake of readability and sacrifices performance. Not a good thing here.